Geometry/EcalCommonData/src/EcalBarrelNumberingScheme.cc

```diff
@@ -27,7 +27,7 @@ uint32_t EcalBarrelNumberingScheme::getUnitID(const EcalBaseNumber& baseNumber)
     return 0;
   }
 
-  const std::string& cryName(baseNumber.getLevelName(0).substr(0, 7));  // name of crystal volume
+  const std::string& cryName(baseNumber.getLevelName(0).substr(7, 14));  // name of crystal volume
```

Comment:
But for old DD there is no namespace, so I think it is better to just remove namespace when present, upstream (in addLevel)

Reply:
@ghugo83 - This was a stopgap solution. Instead of hard-coded numbers, I'd go for a `name()` implementation which would remove both a namespace and a copy number. In any case, please, discuss it with @bsunanda.

Reply:
@ianna Yes I am trying to put a name() function as upstream as possible, which temoves both namespace and copy number as added with dd4hep, to fix the regressions. Right now I have it in addLevel call, and it fixes all name-related regressions in the CMS SimHits, but trying to see if I can put it a bit more upstream.